File: server/models/category_group.rb
DB.create_table? :category_groups do
  uuid :uuid, primary_key: true
  foreign_key :user_uuid, :users, null: false, type: 'uuid'
  String :name, null: false, unique: true
  Integer :priority, null: false, default: 0
  DateTime :created_at, null: false
  DateTime :updated_at, null: false
end

class CategoryGroup < Sequel::Model
  many_to_one :user, key: :user_uuid
end

File: packages/server/src/database/entities/CategoryGroup.js
import {
  Column,
  CreateDateColumn,
  Entity,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm'
import {Category} from './Category'
import {User} from './User'

@Entity()
export class CategoryGroup {
  @PrimaryGeneratedColumn('uuid')
  uuid = undefined

  @ManyToOne(() => User, (user) => user.category_groups, {nullable: false})
  user = undefined

  @OneToMany(() => Category, (category) => category.category_group)
  categories = undefined

  @Column('varchar')
  name = undefined

  @Column('integer')
  priority = undefined

  @CreateDateColumn()
  created_at = undefined

  @UpdateDateColumn()
  updated_at = undefined
}
